File: pythonSlang/step_2/ExpressionBuilder.py
from RDParser import RDParser

class ExpressionBuilder:
    def __init__(self, expr):
        self._expr_string = expr

    def get_expression(self):
        p = RDParser(self._expr_string)
        return p.call_expr()

def main():
    expr_str = "((12 * 12 * 12) + 1) - ((10 * 10 * 10 ) + (9 *9 *9 )))"
    b  = ExpressionBuilder(expr_str)
    e = b.get_expression()
    print (f"Expression in String: {expr_str} ")
    print (f"AST: {e}")
    print (f"Result: { e.evaluate()}")

if __name__ == "__main__":
    main()

File: SlangJava/part04/src/slang4java/builder/ExpressionBuilder.java
package slang4java.builder;

public class ExpressionBuilder extends AbstractBuilder{
    public String _expr_string;

    public ExpressionBuilder(String _expr_string) {
        this._expr_string = _expr_string;
    }

    //
//    public Exp getExpression(){
//        RDPaser p = new RDPaser(_expr_string);
//        return  p.callExpr();
//    }

}
